packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -490,6 +491,69 @@ class _AppBarState extends State<AppBar> {
   }
 }
 
+class _FloatingAppBar extends StatefulWidget {
+  _FloatingAppBar({ Key key, this.child }) : super(key: key);
+
+  final Widget child;
+
+  @override
+  _FloatingAppBarState createState() => new _FloatingAppBarState();
+}
+
+// A wrapper for the widget created by _SliverAppBarDelegate that starts and
+/// stops the floating appbar's snap-into-view or snap-out-of-view animation.
+class _FloatingAppBarState extends State<_FloatingAppBar> {
+  ScrollableState _scrollable;
+  ScrollActivity _activity;
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    _updateScrollPositionListener();
+  }
+
+  @override
+  void dispose() {
+    _scrollable?.position?.removeActivityChangedListener(_scrollPositionListener);
+    super.dispose();
+  }
+
+  void _updateScrollPositionListener() {
+    final ScrollableState scrollable = Scrollable.of(context);
+    _scrollable?.position?.removeActivityChangedListener(_scrollPositionListener);
+    _scrollable = scrollable;
+    _scrollable?.position?.addActivityChangedListener(_scrollPositionListener);
```

Comment:
You won't necessarily rebuild just because the Scrollable rebuilt...  How are you even notified if the whole scrollable changes?  You probably need something like an InheritedWidget here.  These are the problems that InheritedWidget solve.